doc/develop/toolchains/arm_compiler.rst

```diff
@@ -0,0 +1,52 @@
+.. _toolchain_armclang:
+
+Arm Compiler 6
+##############
+
+#. Download and install a development suite containing the `Arm Compiler 6`_
+   for your operating system.
+
+#. :ref:`Set these environment variables <env_vars>`:
+
+   - Set :envvar:`ZEPHYR_TOOLCHAIN_VARIANT` to ``armclang``.
+   - Set :envvar:`ARMCLANG_TOOLCHAIN_PATH` to the toolchain installation
+     directory.
+
+#. The Arm Compiler 6 needs the :envvar:`ARMLMD_LICENSE_FILE` environment
+   variable to point to your license file or server.
+
+For example:
+
+   .. code-block:: console
+
+      # Linux, macOS, license file:
+      export ARMLMD_LICENSE_FILE=/<path>/license_armds.dat
+      # Linux, macOS, license server:
+      export ARMLMD_LICENSE_FILE=8224@myserver
+
+      # Windows, license file:
+      > set ARMLMD_LICENSE_FILE=c:\<path>\license_armds.dat
+      # Windows, license server:
+      > set ARMLMD_LICENSE_FILE=8224@myserver
+
+#. If the Arm Compiler 6 was installed as part of an Arm Development Studio, then
+   you must set the :envvar:`ARM_PRODUCT_DEF` to point to the product definition file:
+   See also: `Product and toolkit configuration <https://developer.arm.com/tools-and-software/software-development-tools/license-management/resources/product-and-toolkit-configuration>`_.
+   For example if the Arm Development Studio is installed in:
+   ``/opt/armds-2020-1`` with a Gold license, then set :envvar:`ARM_PRODUCT_DEF`
```

Comment:
is this `:envvar:` defined somehwere else? ref https://www.sphinx-doc.org/en/master/usage/restructuredtext/domains.html#directive-envvar

toolchain section should likely have a place where all are defined, otherwise this directive doesn't add much value other than marking this as an env var from a source perspective.